script.js

```diff
@@ -1,70 +1,96 @@
-    var requestProductTemplate = $.get('product-template.html').done(function(template) {
+  var _template = function(str, param) {
+    return str.replace(/\{(.*?)\}/g, function(match, token) {
+      return param[token];
+    });
+  }
 
-      var data = products;
-      var template = template;
+  var requestData = $.getJSON('data.json').done(function(productData) {
+    var requestProductTemplate = $.get('product-template.html').done(function(productTemplateHtml) {
 
       function TOMApp() {
-        this.products = [];
-      }
-
-      TOMApp.prototype.getproducts = function() {
         var self = this;
-        for (i=0; i<data.sales.length ; i++) {
-          self.products.push(new productobj(data.sales[i], i)  );
+
+        self.regions = {
+          "$mainRegion": $("#content")
         }
+
+        self.products = [];
+
+        self.initProducts();
+        self.render();
       }
 
-      TOMApp.prototype.updateproducthtml = function() {
-        for( i=0; i< this.products.length ; i++){
-          this.products[i].updatehtml();
+      TOMApp.prototype.initProducts = function() {
+        var self = this;
+
+        for (var i=0; i<productData.sales.length ; i++) {
+          self.products.push(new Product(productData.sales[i], i)  );
         }
       }
 
-      TOMApp.prototype.updatedom = function() {
-        thishtml=''; 
-        for(var i=0; i< this.products.length ; i++){
+      TOMApp.prototype.render = function() {
+        var self = this;
+
+        self.regions.$mainRegion.empty();
+
+        var $row = null;
+        for(var i=0; i< self.products.length ; i++){
           
           if (i % 3 == 0 ) {  
-            thishtml += "<div class='row'>"; 
+            $row = $("<div>").addClass("row");
             console.log("START");
           }
 
-          thishtml += this.products[i].htmlview;
-          if ((i % 3 == 2) || i == (this.products.length-1) ) {
-            thishtml += "</div>";
+          $row.append(self.products[i].render());
+
+          if ((i % 3 == 2) || i == (self.products.length-1)) {
+            self.regions.$mainRegion.append($row);
             console.log("FINISH");
           }
         }
-        $("#content").append(thishtml)
       }
 
-      function productobj(product, i) {
-        this.photo        = product.photos.medium_half;
-        this.title        = product.name;
-        this.tagline      = product.tagline;
-        this.url          = product.url;
-        this.htmlview     = "";
-        this.index        = i;
-        this.custom_class = "col"+ ((i % 3) +1);
+      function Product(product, i) {
+        var self = this;
+
+        self.photo        = product.photos.medium_half;
+        self.title        = product.name;
+        self.tagline      = product.tagline;
+        self.url          = product.url;
+        self.index        = i;
+        self.custom_class = "col"+ ((i % 3) +1);
+
+        self.$elem = null;
```

Comment:
`Product` class is a representation of model/view. Holding the actual DOM instead of the html string established the relationship between page DOM and the object instance of `Product`. This makes data-binding feasible because we could access the DOM via its representation of `Product` and attach/detach events. Inspired by Backbone views.